### ISSUE TITLE
Accessibility: Visual label missing - aria-label present

### ISSUE THREAD

**Issue by vikrampm1** [woothemes/woocommerce#52129]
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:

The visual labels for these input fields rely only on the default text in the field. Screen reader users have access to an aria-label that other users do not have access to, that fields are for minimum and maximum prices. Other users are required to intuit the purpos of the field based on the numbers in them and their visual placement below the slider line. 

The lack of visible and descriptive labels might cause difficulties for all users when trying to enter information—but particularly for those with cognitive, language, and learning disabilities.

![Image](https://github.com/user-attachments/assets/abd3a873-f2bf-4db4-8646-131e2da41382)

### Component or Page: 

Component: Product Archive Price Filter Block     

### Name affected element:

Inputs for price filtering, such as:
1. Filter products by minimum price
2. Filter products by maximum price

### Relevant code snippets:

```
<input class=""wc-block-formatted-money-amount wc-block-components-formatted-money-amount wc-block-price-filter__amount wc-block-price-filter__amount--min wc-block-form-text-input wc-block-components-price-slider__amount wc-block-components-price-slider__amount--min"" aria-label=""Filter products by minimum price"" type=""text"" value=""$10"" inputmode=""numeric"">

<input class="wc-block-formatted-money-amount wc-block-components-formatted-money-amount wc-block-price-filter__amount wc-block-price-filter__amount--max wc-block-form-text-input wc-block-components-price-slider__amount wc-block-components-price-slider__amount--max" aria-label="Filter products by maximum price"" type="text" value="$405" inputmode=""numeric">
```

### Recommended fix:

Provide a label for each input, likely below the inputs. Add a <label> tag and use a "for" attribute to link the label to the input. Ensure the label is clear and descriptive. I.e., "Min. Price" and "Max. Price"
